src/Common/tests/System/Buffers/NativeOwnedMemory.cs

```diff
@@ -0,0 +1,106 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    internal sealed class NativeOwnedMemory : OwnedMemory<byte>
+    {
+        private readonly int _length;
+        private IntPtr _ptr;
+        private int _retainedCount;
+        private bool _disposed;
+
+        public NativeOwnedMemory(int length)
+        {
+            _length = length;
+            _ptr = Marshal.AllocHGlobal(length);
```

Comment:
It might be good to add a finalizer to this type and delete the memory plus optionally log that the object should be disposed.

Reply:
Ok.  But let's delete this in the near future by adding something public and real 😛